shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.js

```diff
@@ -348,9 +180,182 @@ exports.serveCaseToIrsInteractor = async (
+    'address1',
+    'address2',
+    'address3',
+    'city',
+    'state',
+    'postalCode',
+  ];
+
+  const contactAddressesAreDifferent = Object.keys(contactInformationDiff).some(
+    field => addressFields.includes(field),
+  );
+
+  let secondaryPdfData;
+
+  if (contactAddressesAreDifferent) {
+    secondaryPdfData = await applicationContext
+      .getDocumentGenerators()
+      .noticeOfReceiptOfPetition({
+        applicationContext,
+        data: {
+          address: contactSecondary,
+          caseCaptionExtension,
+          caseTitle,
+          docketNumberWithSuffix,
+          preferredTrialCity,
+          receivedAtFormatted: applicationContext
+            .getUtilities()
+            .formatDateString(receivedAt, 'MMMM D, YYYY'),
+          servedDate: applicationContext
+            .getUtilities()
+            .formatDateString(caseEntity.getIrsSendDate(), 'MMMM D, YYYY'),
+        },
+      });
+
+    const { PDFDocument } = await applicationContext.getPdfLib();
+    const pdfDoc = await PDFDocument.load(pdfData);
+    const secondaryPdfDoc = await PDFDocument.load(secondaryPdfData);
+    const coverPageDocumentPages = await pdfDoc.copyPages(
+      secondaryPdfDoc,
+      secondaryPdfDoc.getPageIndices(),
+    );
+    pdfDoc.insertPage(1, coverPageDocumentPages[0]);
+
+    const pdfDataBuffer = await pdfDoc.save();
+    return Buffer.from(pdfDataBuffer);
+  }
+};
+
+/**
+ * serveCaseToIrsInteractor
+ *
+ * @param {object} applicationContext the application context
+ * @param {object} providers the providers object
+ * @param {string} providers.docketNumber the docket number of the case
+ * @returns {Buffer} paper service pdf if the case is a paper case
+ */
+exports.serveCaseToIrsInteractor = async (
+  applicationContext,
+  { docketNumber },
+) => {
+  const user = applicationContext.getCurrentUser();
+
+  if (!isAuthorized(user, ROLE_PERMISSIONS.SERVE_PETITION)) {
+    throw new UnauthorizedError('Unauthorized');
+  }
+
+  const caseToBatch = await applicationContext
+    .getPersistenceGateway()
+    .getCaseByDocketNumber({
+      applicationContext,
+      docketNumber,
+    });
+
+  let caseEntity = new Case(caseToBatch, { applicationContext });
+
+  caseEntity.markAsSentToIRS();
+
+  if (caseEntity.isPaper) {
+    addDocketEntries({ caseEntity });
+  }
+
+  for (const initialDocumentTypeKey of Object.keys(INITIAL_DOCUMENT_TYPES)) {
+    await applicationContext.getUtilities().serveCaseDocument({
+      applicationContext,
+      caseEntity,
+      initialDocumentTypeKey,
+    });
+  }
+
+  addDocketEntryForPaymentStatus({
+    applicationContext,
+    caseEntity,
+    user,
+  });
+
+  caseEntity
+    .updateCaseCaptionDocketRecord({ applicationContext })
+    .updateDocketNumberRecord({ applicationContext })
+    .validate();
+
+  await createPetitionWorkItems({
+    applicationContext,
+    caseEntity,
+    user,
+  });
+
+  // TODO: figure out why we need this for making the petitioner become served
```

Comment:
is this tracked elsewhere?

Reply:
this is actually old code.  @rachelschneiderman and I worked on refactoring this function and we managed to get it working without needing to call updateCase twice.  I'll remove the comment.